### PR TITLE
⏪️ Revert software.amazon.awssdk:apache-client dependency override

### LINKED ISSUE #3366: [Quarkus 2.6.0.Final] Remove `software.amazon.awssdk:apache-client` dependency override
In Quarkus 2.6.0.CR1 there's an invalid dependency declaration in `quarkus-bom` for the AWS `apache-client`.

So for now, I've worked around it by specifying the dependency in our own BOM.

When 2.6.0.Final is available, we should be able to remove it.

### PR DESCRIPTION
**Pull request by zbendhiba**
Fixes #3366

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: poms/bom/pom.xml [hunk 1]
```diff
@@ -6055,18 +6048,6 @@
...
-                <groupId>software.amazon.awssdk</groupId>
-                <artifactId>apache-client</artifactId>
-                <version>${awssdk.version}</version>
-                <exclusions>
-                    <exclusion>
-                        <groupId>commons-logging</groupId>
-                        <artifactId>commons-logging</artifactId>
-                    </exclusion>
-                </exclusions>
-            </dependency>
```
Comment (ppalaga) [inline poms/bom/pom.xml:6069]: I think this is the only part we should remove now that we upgrade to Quarkus 2.6. I think `software.amazon.awssdk:bom` in version synced with quarkiverse amazonservices is still needed to keep the versions of software.amazon.awssdk dependencies consistent.

Comment (aldettinger) [reply poms/bom/pom.xml:6069]: Ok, my note in #3366 is misleading then. Sorry about that.

Comment (zbendhiba) [reply poms/bom/pom.xml:6069]: ok, let me change this.

Comment (zbendhiba) [reply poms/bom/pom.xml:6069]: should be fine by now

**Review by aldettinger — APPROVED** [review]

**Review by ppalaga — APPROVED** [review]
Thanks @zbendhiba!

**Comment by zbendhiba** [comment]
I can't reproduce those errors locally. As it is AWS2 related, it's maybe related to these changes ?

**Comment by ppalaga** [comment]
I can try to reproduce

**Comment by aldettinger** [comment]
`Discovered unresolved method during parsing: org.apache.commons.logging.impl.LogFactoryImpl.handleThrowable(java.lang.Throwable)`
A native build issue, I would expect it to be reproducible.

@zbendhiba Are you building locally with native-image 21.3.0 as CI does ?

**Comment by ppalaga** [comment]
Replaced by https://github.com/apache/camel-quarkus/pull/3406

**Pull request closed without merging**